### PR TITLE
Added scorep

### PR DESCRIPTION
**Pull request by nvarini**

### CODE REVIEW

File: paien.yaml [hunk 1]
```diff
@@ -293,7 +293,7 @@ packages:
       - hdf5@1.10.1+szip+mpi+hl+fortran+cxx
       - scotch@6.0.4+esmumps+mpi~metis
 # TODO: update scorep as described in issue SL-557
```
Comment (rmsds) [inline paien.yaml]: Could you also delete this comment? I think your commit addresses the issue in question.

**Review by rmsds — DISMISSED** [review]

**Review by alalazo — CHANGES_REQUESTED** [review]
Looking at Jenkins CI, it seems that `pdt-toolkit` fails to build (and `scorep` is then installed accidentally without it). Can you fix this? Also: can you make sure we are using the same Qt 5.10.0 that is installed on the clusters?

**Review by alalazo — CHANGES_REQUESTED** [review]
Almost there. It seems installation is now failing only for `gcc@7.3.0`:
```console
...
==> Installing scorep
Reversed (or previously applied) patch detected!  Assume -R? [n] 
Apply anyway? [n] 
1 out of 1 hunk ignored -- saving rejects to file src/adapters/compiler/gcc-plugin/scorep_plugin_inst_descriptor.inc.c.rej
Reversed (or previously applied) patch detected!  Assume -R? [n] 
Apply anyway? [n] 
1 out of 1 hunk ignored -- saving rejects to file src/adapters/compiler/gcc-plugin/scorep_plugin_tree-flow.h.rej
==> Using cached archive: /home/scitasbuild/paien/pr/spack.DGgpu/var/spack/cache/scorep/scorep-4.1.tar.gz
==> Staging archive: /home/scitasbuild/paien/pr/spack.DGgpu/var/spack/stage/scorep-4.1-uy2canzhl34vr4xzrmjt63zc5vadywqi/scorep-4.1.tar.gz
==> Created stage in /home/scitasbuild/paien/pr/spack.DGgpu/var/spack/stage/scorep-4.1-uy2canzhl34vr4xzrmjt63zc5vadywqi
==> Patch gcc7.patch failed.
==> Error: ProcessError: Command exited with status 1:
    '/usr/bin/patch' '-s' '-p' '1' '-i' '/home/scitasbuild/paien/pr/spack.DGgpu/var/spack/repos/builtin/packages/scorep/gcc7.patch' '-d' '.'
...
```
because of a patch that couldn't / shouldn't be applied?

**Comment by alalazo** [comment]
Just checked the result of Jenkins, and I couldn't see any error. Every architecture installs 5 versions of `scorep`, which is what we want. Trying to merge it and see how it deploys.

**Review by alalazo — APPROVED** [review]

**Pull request merged**